src/Stratis.Bitcoin.Features.Wallet/WalletTransactionHandler.cs

```diff
@@ -380,146 +400,128 @@ protected void AddOpReturnOutput(TransactionBuildContext context)
-        /// </summary>
-        /// <remarks>
-        /// A Bitcoin has to spend the entire UTXO, if total value is greater then the send target
-        /// the rest of the coins go in to a change address that is under the senders control.
-        /// </remarks>
-        public HdAddress ChangeAddress { get; set; }
-
-        /// <summary>
-        /// The total fee on the transaction.
-        /// </summary>
-        public Money TransactionFee { get; set; }
-
-        /// <summary>
-        /// The final transaction.
-        /// </summary>
-        public Transaction Transaction { get; set; }
-
-        /// <summary>
-        /// The password that protects the wallet in <see cref="WalletAccountReference"/>.
-        /// </summary>
-        /// <remarks>
-        /// TODO: replace this with System.Security.SecureString (https://github.com/dotnet/corefx/tree/master/src/System.Security.SecureString)
-        /// More info (https://github.com/dotnet/corefx/issues/1387)
-        /// </remarks>
-        public string WalletPassword { get; set; }
-
-        /// <summary>
-        /// The inputs that must be used when building the transaction.
-        /// </summary>
-        /// <remarks>
-        /// The inputs are required to be part of the wallet.
-        /// </remarks>
-        public List<OutPoint> SelectedInputs { get; set; }
-
-        /// <summary>
-        /// If false, allows unselected inputs, but requires all selected inputs be used.
-        /// </summary>
-        public bool AllowOtherInputs { get; set; }
-
-        /// <summary>
-        /// Specify whether to sign the transaction.
-        /// </summary>
-        public bool Sign { get; set; }
-
-        /// <summary>
-        /// Allows the context to specify a <see cref="FeeRate"/> when building a transaction.
-        /// </summary>
-        public FeeRate OverrideFeeRate { get; set; }
-
-        /// <summary>
-        /// Shuffles transaction inputs and outputs for increased privacy.
-        /// </summary>
-        public bool Shuffle { get; set; }
-
-        /// <summary>
-        /// Optional data to be added as an extra OP_RETURN transaction output with Money.Zero value.
-        /// </summary>
-        public string OpReturnData { get; set; }
-    }
-
-    /// <summary>
-    /// Represents recipients of a payment, used in <see cref="WalletTransactionHandler.BuildTransaction"/>.
-    /// </summary>
-    public class Recipient
-    {
-        /// <summary>
-        /// The destination script.
-        /// </summary>
-        public Script ScriptPubKey { get; set; }
+            public TransactionBuildContext(Network network, TransactionBuildOptions options)
+            {
+                this.TransactionBuilder = new TransactionBuilder(network);
+                this.BuildOptions = options;
+
+                this.AccountReference = options.WalletAccountReference;
+                this.Recipients = options.Recipients;
```

Comment:
I see the guard is removed on this now, it is because it can't be null inside `options`, right?